src/south/ProtocolHandler.class.js

```diff
@@ -75,7 +108,24 @@ class ProtocolHandler {
   }
 
   async connect() {
-    const { dataSourceId, protocol } = this.dataSource
+    const { dataSourceId, protocol, startTime } = this.dataSource
+
+    if (this.supportedModes?.supportHistory) {
+      // Initialize lastCompletedAt for every scanMode
+      // "startTime" is currently a "hidden" parameter of oibus.json
+      const defaultLastCompletedAt = startTime ? new Date(startTime) : new Date()
+
+      // Disable ESLint check because we need for..of loop to support async calls
+      // eslint-disable-next-line no-restricted-syntax
+      for (const scanMode of Object.keys(this.lastCompletedAt)) {
+        // Disable ESLint check because we want to get the values one by one to avoid parallel access to the SQLite database
+        // eslint-disable-next-line no-await-in-loop
+        const lastCompletedAt = await this.getConfig(`lastCompletedAt-${scanMode}`)
+        this.lastCompletedAt[scanMode] = lastCompletedAt ? new Date(lastCompletedAt) : defaultLastCompletedAt
+        this.logger.info(`Initializing lastCompletedAt for ${scanMode} with ${this.lastCompletedAt[scanMode]}`)
+      }
+    }
```

Comment:
Shouldn't this block be in the constructor instead of the connect ? In the if block l.74

Reply:
It is not possible to use await in constructor.
(Using await requires the function to be async. If the function is async it returns a Promise. The constructor returns the newly created object)